src/solidity/astHelper.js

```diff
@@ -0,0 +1,81 @@
+'use strict'
+var AstWalker = require('../util/astWalker')
+
+/**
+  * return all contract definitions of the given @astList
+  *
+  * @param {Object} astList  - AST nodes of all the sources
+  * @return {Object} - return the AST node of found contracts
+  */
+function getContractsDefinition (astList) {
+  var contracts = {}
+  var walker = new AstWalker()
+  walker.walkAstList(astList, { 'ContractDefinition': function (node) {
+    contracts[node.id] = node
+    return false
+  }})
+  return contracts
+}
+
+/**
+  * returns the linearized base contracts of the contract @arg id
+  *
+  * @param {Int} id  - contract id for which based contracts should be resolved
+  * @param {Map} contracts  - all contracts defined in the current context
+  * @return {Object} - return the AST node of found contract
```

Comment:
`@return {Array} array of base contracts in derived to base order as AST nodes.`

Please also clarify whether it only returns the direct base contracts or the full list of linearized base contracts.